bpaotu/bpaotu/biom.py

```diff
@@ -48,17 +49,20 @@ def biom_header():
 
 def otu_rows(query, otu_to_row):
     q = query.matching_otus()
-    fields = ('amplicon', 'kingdom', 'phylum', 'class', 'order', 'family', 'genus', 'species')
-    for idx, otu in enumerate(q.yield_per(50)):
-        otu_to_row[otu.id] = idx
+    taxonomy_fields = ('kingdom', 'phylum', 'class', 'order', 'family', 'genus', 'species')
 
+    for idx, otu in enumerate(q.yield_per(50)):
         def get_value(attr):
             if attr == 'class':
                 attr = 'klass'
             return val_or_empty(getattr(otu, attr))
 
-        metadata = ','.join(k_v(f, get_value(f)) for f in fields if get_value(f) != '')
-        yield '{"id": "%s","metadata": {%s}}' % (otu.code, metadata)
+        otu_to_row[otu.id] = idx
+        taxonomy_array = [get_value(f) for f in taxonomy_fields]
+        yield '{"id": "%s","metadata": {%s,%s}}' % (
+            otu.code,
+            k_v("amplicon", get_value("amplicon")),
```

Comment:
Minor: use single-quote for string literals to be consistent with the rest of the file.